src/nevermined/Nfts.ts

```diff
@@ -512,4 +513,78 @@ export class Nfts extends Instantiable {
             throw Error(`Error listing ${ddo.shortId()} on secondary markets`)
         }
     }
+
+    public async lockBuyersPayments(
+        buyer: string,
+        seller: string,
+        nftPrice: number,
+        ddo: DDO
+    ) {
+        const buyersAccount = new Account(buyer)
+        const { token } = this.nevermined.keeper
+
+        const agreementId = utils.generateId()
+        const assetRewards = new AssetRewards(new Map([[seller, nftPrice]]))
+
+        const scale = 10 ** (await token.decimals())
+        await buyersAccount.requestTokens(nftPrice / scale)
+
+        const receipt = await this.nevermined.agreements.conditions.lockPayment(
+            agreementId,
+            ddo.id,
+            assetRewards.getAmounts(),
+            assetRewards.getReceivers(),
+            token.getAddress(),
+            buyersAccount
+        )
+
+        if (!receipt) throw new Error('Transaction Failed.')
+    }
+
+    public async transferNftToBuyer(
+        seller: string,
+        buyer: string,
+        ddo: DDO,
+        nftPrice: number
+    ) {
+        const sellerAccount = new Account(seller)
+        const agreementId = utils.generateId()
+
+        const assetRewards = new AssetRewards(new Map([[buyer, nftPrice]]))
+
+        const receipt = await this.nevermined.agreements.conditions.transferNft(
+            agreementId,
+            ddo,
+            assetRewards.getAmounts(),
+            assetRewards.getReceivers(),
+            1,
```

Comment:
It is debatable if we want to enable single buys or should come from a param, but for the current use case it is one for sure.

Reply:
I will give a default value of one if its not provided